File: src/test/java/kitchenpos/application/fake/FakeOrderRepository.java
package kitchenpos.application.fake;

import kitchenpos.domain.Order;
import kitchenpos.domain.OrderRepository;
import kitchenpos.domain.OrderStatus;
import kitchenpos.domain.OrderTable;

import java.util.*;
import java.util.stream.Collectors;

public class FakeOrderRepository implements OrderRepository {
    Map<UUID, Order> memory = new HashMap<>();
    @Override
    public Order save(Order order) {
        memory.put(order.getId(), order);
        return memory.get(order.getId());
    }

    @Override
    public Optional<Order> findById(UUID id) {
        return Optional.ofNullable(memory.get(id));
    }

    @Override
    public List<Order> findAll() {
        return memory.values().stream().collect(Collectors.toList());
    }

    @Override
    public boolean existsByOrderTableAndStatusNot(OrderTable orderTable, OrderStatus status){
        return memory.values()
                .stream()
                .anyMatch(order -> order.getOrderTable().equals(orderTable) && order.getStatus().equals(status));
    }
}

File: src/test/kotlin/kitchenpos/application/fake/FakeOrderRepository.kt
package kitchenpos.application.fake

import kitchenpos.domain.Order
import kitchenpos.domain.OrderRepository
import kitchenpos.domain.OrderStatus
import kitchenpos.domain.OrderTable
import java.util.*

class FakeOrderRepository : OrderRepository {
    private val orders = mutableMapOf<UUID, Order>()

    override fun existsByOrderTableAndStatusNot(
        orderTable: OrderTable,
        status: OrderStatus?,
    ): Boolean {
        return orders.any {
            it.value.orderTable?.id == orderTable.id && it.value.status != status
        }
    }

    override fun save(order: Order): Order {
        orders[order.id] = order
        return order
    }

    override fun findById(id: UUID?): Optional<Order> {
        return Optional.ofNullable(orders[id])
    }

    override fun findAll(): MutableList<Order> {
        return orders.values.toMutableList()
    }
}
